api/envoy/config/route/v3/route_components.proto

```diff
@@ -964,11 +975,18 @@ message RouteAction {
   // route.
   repeated RateLimit rate_limits = 13;
 
-  // Specifies if the rate limit filter should include the virtual host rate
-  // limits. By default, if the route configured rate limits, the virtual host
-  // :ref:`rate_limits <envoy_api_field_config.route.v3.VirtualHost.rate_limits>` are not applied to the
-  // request.
-  google.protobuf.BoolValue include_vh_rate_limits = 14;
+  oneof vh_rate_limits_options {
+    // Specifies if the rate limit filter should include the virtual host rate
+    // limits. By default, if the route configured rate limits, the virtual host
+    // :ref:`rate_limits <envoy_api_field_config.route.v3.VirtualHost.rate_limits>` are not applied to the
+    // request.
+    //
+    // This field is deprecated. Please use :ref:`vh_rate_limits <envoy_v3_api_field_config.route.v3.RouteAction.vh_rate_limits>`
+    google.protobuf.BoolValue include_vh_rate_limits = 14 [deprecated = true];
```

Comment:
If we are going to deprecate this field, should we just move this config to per-route config for the rate limit filter? All of this pre-dates the per-filter per-route config, and it should really go there. I would rather not add new debt here. WDYT?

Reply:
Just to clarify do you mean putting it under this [route config](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-msg-config-route-v3-route)?

Reply:
What I mean is something like this: https://github.com/envoyproxy/envoy/blob/670b2ebb927f68b6468326a54cf34be94d1ca31a/api/envoy/config/filter/http/ext_authz/v2/ext_authz.proto#L208

If you take a look at the filter code, we have the ability to have a proto/config on a per-route/virtual host basis. This is the new way of doing what is done here, which is to effectively inline filter specific config in the main route protos. If we are going to deprecate, I would much rather move to the per-filter config model.

Reply:
Ok, thanks for clarifying! I'll move it towards the per-filter config model